src/pe_reports/metrics.py

```diff
@@ -12,25 +22,571 @@ def __init__(self, start_date, end_date, org_uid):
+        return risky_assets
+
+    def risky_ports_count(self):
+        """Return total count of insecure protocols."""
+        insecure_df = self.insecure_df
+        insecure = insecure_df[insecure_df["type"] == "Insecure Protocol"]
+        insecure = insecure[
+            (insecure["protocol"] != "http") & (insecure["protocol"] != "smtp")
+        ]
+        risky_assets = insecure[["ip", "protocol"]].drop_duplicates(keep="first")
+
+        pro_count = risky_assets.groupby(["protocol"], as_index=False)["protocol"].agg(
+            {"id_count": "count"}
+        )
+
+        # Total Open Ports with Insecure protocols
+        return pro_count["id_count"].sum()
+
+    def total_verif_vulns(self):
+        """Return total count of verified vulns."""
+        vulns_df = self.vulns_df
+        verif_vulns = (
+            vulns_df[["cve", "ip", "port"]]
+            .groupby("cve")
+            .agg(lambda x: "  ".join(set(x)))
+            .reset_index()
+        )
+
+        if len(verif_vulns) > 0:
+            verif_vulns["count"] = verif_vulns["ip"].str.split("  ").str.len()
+            verifVulns = verif_vulns["count"].sum()
+
+        else:
+            verifVulns = 0
+
+        return verifVulns
+
+    def unverified_cve(self):
+        """Return top 15 unverified cves and their counts."""
+        insecure_df = self.insecure_df
+        unverif_df = insecure_df[insecure_df["type"] != "Insecure Protocol"]
+        unverif_df = unverif_df.copy()
+        unverif_df["potential_vulns"] = (
+            unverif_df["potential_vulns"].sort_values().apply(lambda x: sorted(x))
+        )
+        unverif_df["potential_vulns"] = unverif_df["potential_vulns"].astype("str")
+        unverif_df = (
+            unverif_df[["potential_vulns", "ip"]]
+            .drop_duplicates(keep="first")
+            .reset_index(drop=True)
+        )
+        unverif_df["count"] = unverif_df["potential_vulns"].str.split(",").str.len()
+        unverif_df = unverif_df[["ip", "count"]]
+        unverif_df = unverif_df.sort_values(by=["count"], ascending=False)
+        unverif_df = unverif_df[:15].reset_index(drop=True)
+        return unverif_df
+
+    def unverified_vuln_count(self):
+        """Return the count of ips with unverified vulnerabilities."""
+        insecure_df = self.insecure_df
+        unverif_df = insecure_df[insecure_df["type"] != "Insecure Protocol"]
+        unverif_df = unverif_df.copy()
+        unverif_df["potential_vulns"] = (
+            unverif_df["potential_vulns"].sort_values().apply(lambda x: sorted(x))
+        )
+        unverif_df["potential_vulns"] = unverif_df["potential_vulns"].astype("str")
+        unverif_df = (
+            unverif_df[["potential_vulns", "ip"]]
+            .drop_duplicates(keep="first")
+            .reset_index(drop=True)
+        )
+
+        return len(unverif_df.index)
+
+    def verif_vulns(self):
+        """Return a dataframe with each CVE, the associated ips and the affected ports."""
```

Comment:
Use proper capitalization for acronyms in docstrings:
```suggestion
        """Return a dataframe with each CVE, the associated IPs and the affected ports."""
```